Refresh.GameServer/Endpoints/Game/Levels/FilterSettings/LevelFilterSettings.cs

```diff
@@ -0,0 +1,120 @@
+    /// The game(s) to filter the results to
+    /// </summary>
+    public GameFilterType GameFilterType = GameFilterType.Both;
+    /// <summary>
+    /// Whether or not to display Lbp1 levels in the results (corresponds to gameFilter[]=lbp1)
+    /// </summary>
+    public bool DisplayLbp1 = true;
+    /// <summary>
+    /// Whether or not to display Lbp2 levels in the results (corresponds to gameFilter[]=lbp2)
+    /// </summary>
+    public bool DisplayLbp2 = true;
+    /// <summary>
+    /// Whether or not to display Lbp3 levels in the results (corresponds to gameFilter[]=lbp3)
+    /// </summary>
+    public bool DisplayLbp3 = true;
+    /// <summary>
+    /// Whether or not the user's own levels should be excluded from the results
+    /// </summary>
+    public bool ExcludeMyLevels = false;
+    /// <summary>
+    /// The amount of players to filter for
+    /// eg. only show levels which are compatible with that many players
+    /// </summary>
+    public byte Players = 0;
+    /// <summary>
+    /// The filter setting for move levels
+    /// </summary>
+    public MoveFilterType MoveFilterType = MoveFilterType.True;
+    /// <summary>
+    /// The first label filter
+    /// </summary>
+    public string? LabelFilter0 = null;
+    /// <summary>
+    /// The second label filter
+    /// </summary>
+    public string? LabelFilter1 = null;
+    /// <summary>
+    /// The third label filter
+    /// </summary>
+    public string? LabelFilter2 = null;
+
+    public LevelFilterSettings()
+    {
+    }
+    
+    /// <summary>
+    /// Gets the filter settings from a request
+    /// </summary>
+    /// <param name="context"></param>
+    public LevelFilterSettings(RequestContext context)
+    {
+        string[]? gameFilters = context.QueryString.GetValues(context.IsApi() ? "gameFilter" : "gameFilter[]");
+        if (gameFilters != null)
+        {
+            this.DisplayLbp1 = false;
+            this.DisplayLbp2 = false;
+            this.DisplayLbp3 = false;
+            foreach (string gameFilter in gameFilters)
+                switch (gameFilter)
+                {
+                    case "lbp1":
+                        this.DisplayLbp1 = true;
+                        break;
+                    case "lbp2":
+                        this.DisplayLbp2 = true;
+                        break;
+                    case "lbp3":
+                        this.DisplayLbp3 = true;
+                        break;
+                }
+        }
+        
+        string? gameFilterType = context.QueryString.Get("gameFilterType");
+        if (gameFilterType != null)
+            this.GameFilterType = gameFilterType switch
+            {
+                "lbp1" => GameFilterType.LittleBigPlanet1,
+                "lbp2" => GameFilterType.LittleBigPlanet2,
+                "both" => GameFilterType.Both,
+                //NOTE: We dont throw an exception here because an unknown filter setting could very well just be an unhandled case,
```

Comment:
Ensuring everything stays working is more than acceptable, but I still think we should at least log a warning in this case.

Reply:
What would be the proper way to access the logger from this code? pass it in through the constructor?

Reply:
Good point... I'm not exactly sure. Requiring a logger in the constructor would certainly be sloppy, so I'd say store it statically, but something about that irks me.

It might be okay to just see if this catches on fire. I don't think this will be an issue.